plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -962,7 +1004,58 @@ void producer_plugin::plugin_initialize(const boost::program_options::variables_
       }
    }
 
-   my->_max_read_only_transaction_time_ms = options.at("max-read-only-transaction-time").as<int32_t>();
+   my->_ro_thread_pool_size = options.at( "read-only-threads" ).as<uint16_t>();
+   // only initialize other read-only options when read-only thread pool is enabled
+   if ( my->_ro_thread_pool_size > 0 ) {
+      EOS_ASSERT( !my->_production_enabled, plugin_config_exception, "--read-only-thread does not work when production is enabled" );
```

Comment:
`EOS_ASSERT( my->_producers.empty(), "--read-only-thread not allowed on producer node" );`

Reply:
Sorry, I missed the first part, thinking my error text was not concise.